packages/flutter/lib/src/widgets/editable_text_context_menu_button_items_builder.dart

```diff
@@ -0,0 +1,296 @@
+class EditableTextContextMenuButtonItemsBuilder extends StatefulWidget {
+  /// Creates an instance of [EditableTextContextMenuButtonItemsBuilder].
+  const EditableTextContextMenuButtonItemsBuilder({
+    super.key,
+    TargetPlatform? targetPlatform,
+    required this.builder,
+    required this.editableTextState,
+  }) : _targetPlatform = targetPlatform;
+
+  /// Called with a list of [ContextMenuButtonItem]s so the context menu can be
+  /// built.
+  final ToolbarButtonWidgetBuilder builder;
+
+  /// The EditableTextState for the field that will display the text selection
+  /// toolbar.
+  final EditableTextState editableTextState;
+
+  final TargetPlatform? _targetPlatform;
+
+  /// The platform to base the button items on.
+  TargetPlatform get targetPlatform => _targetPlatform ?? defaultTargetPlatform;
+
+  /// Returns true if the given [EditableTextState] supports cut.
+  static bool canCut(EditableTextState editableTextState) {
+    return !editableTextState.widget.readOnly
+        && !editableTextState.widget.obscureText
+        && !editableTextState.textEditingValue.selection.isCollapsed;
+  }
+
+  /// Returns true if the given [EditableTextState] supports copy.
+  static bool canCopy(EditableTextState editableTextState) {
+    return !editableTextState.widget.obscureText
+        && !editableTextState.textEditingValue.selection.isCollapsed;
+  }
+
+  /// Returns true if the given [EditableTextState] supports paste.
+  static bool canPaste(EditableTextState editableTextState) {
+    return !editableTextState.widget.readOnly
+        && (editableTextState.clipboardStatus == null
+          || editableTextState.clipboardStatus!.value == ClipboardStatus.pasteable);
+  }
+
+  /// Returns true if the given [EditableTextState] supports select all.
+  ///
+  /// If [targetPlatform] is not provided, [defaultTargetPlatform] will be used.
+  static bool canSelectAll(EditableTextState editableTextState, [TargetPlatform? targetPlatform]) {
+    if (!editableTextState.widget.enableInteractiveSelection
+        || (editableTextState.widget.readOnly
+            && editableTextState.widget.obscureText)) {
+      return false;
+    }
+
+    switch (targetPlatform ?? defaultTargetPlatform) {
+      case TargetPlatform.macOS:
+        return false;
+      case TargetPlatform.iOS:
+        return editableTextState.textEditingValue.text.isNotEmpty
+            && editableTextState.textEditingValue.selection.isCollapsed;
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+        return editableTextState.textEditingValue.text.isNotEmpty
+           && !(editableTextState.textEditingValue.selection.start == 0
+               && editableTextState.textEditingValue.selection.end == editableTextState.textEditingValue.text.length);
+    }
+  }
+
+  /// Returns the [ContextMenuButtonItem]s for the given [ToolbarOptions].
+  @Deprecated(
```

Comment:
Is this deprecated by accident?

Reply:
No, see the section in my PR description called "New stuff immediately deprecated".  This was created to keep both the old and new stuff working while they both coexist.  Once the deprecations are removed this should go away too.